### ISSUE TITLE
Error responses to add group API requests are inconsistent

### ISSUE THREAD

**Issue by phil-davis** [owncloud/core#40164]
### Steps to reproduce
The webUI uses `/index.php/settings/users/groups` endpoint to "do its thing", rather than the Provisioning API https://doc.owncloud.com/server/next/developer_manual/core/apis/provisioning-api.html#add-group-example

1. On the webUI, try to add a group with a space at the end of the name. Inspect the response in the browser.
2. Create a group with the Provisioning API (and do not refresh the webUI)
```
$ curl -X POST http://admin:admin@172.17.0.1:8080/ocs/v1.php/cloud/groups -d groupid="somegroup"
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>ok</status>
  <statuscode>100</statuscode>
  <message/>
 </meta>
 <data/>
</ocs>
```

3. On the webUI, try to add a group with that name `somegroup`. Inspect the response in the browser.

### Expected behaviour
Both error responses have the same format.

### Actual behaviour
Response 1:
```
{"status":"error","data":{"message":"Unable to add group."}}
```

Response 2:
```
{"message":"Group already exists."}
```

One response has the message inside "data", the other has the message not inside "data".
This makes it difficult for the webUI to correct parse the error response and show it to the user.

### Server configuration
Current core master

**Comment by phil-davis** [comment]
This was discovered due to issue https://github.com/owncloud/user_ldap/issues/742 happening. That is not actually an LDAP problem, but is this core problem.

**Comment by jnweiger** [comment]
Confirmed fixed in 10.11.0-rc.1
1. `{"status":"error","message":"Unable to add group."}`
2. `{"status":"error","message":"Group already exists."}`
